### PR TITLE
Aman/sort filter

### PR DESCRIPTION
**Pull request by ataddesse**
# Description

Please include a summary of the feature or bug and the changes/implementations to tackle it. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by ataddesse** [comment]
I have started using an html attribute instead of felx-box material properties, because that is more recommended by the Angular dev team.

File: frontend/src/app/app.module.ts [hunk 1]
```diff
@@ -54,7 +55,8 @@ import { LeadService } from './services/lead.service';
     MatSidenavModule,
     MatToolbarModule,
     MatTableModule,
-    MatPaginatorModule
+    MatPaginatorModule,
+    MatSortModule
   ],
   providers: [
     LeadService,
```
Comment (r0ddy) [inline frontend/src/app/app.module.ts:62]: Remove the LeadService provider here since it's already provided at the root level.

Comment (ataddesse) [reply frontend/src/app/app.module.ts]: Ight

File: frontend/src/app/crm/leads/leads.component.ts [hunk 2]
```diff
@@ -22,16 +24,25 @@ export class LeadsComponent implements OnInit {
    dataSource : MatTableDataSource<Lead>;
 
    @ViewChild(MatPaginator, {static: true}) paginator: MatPaginator;
+   @ViewChild(MatSort, {static: true}) sort: MatSort;
+
    constructor(private leadService : LeadService) { }
 
    ngOnInit() : void {
-      this.getAllLeads();
+         this.getAllLeads();
```
Comment (r0ddy) [inline frontend/src/app/crm/leads/leads.component.ts:32]: Remove indentation.

Comment (ataddesse) [reply frontend/src/app/crm/leads/leads.component.ts]: Ight

File: frontend/src/app/crm/leads/leads.component.html [hunk 3]
```diff
@@ -3,8 +3,11 @@
 <li *ngFor="let lead of leads">
   <p>{{lead.date}}, {{lead.lead_id}}, {{lead.campaign_id}}, {{lead.gcl_id}}, {{lead.api_version}}, {{lead.form_id}}, {{lead.google_key}}, {{lead.column_data.FULL_NAME}}, {{lead.column_data.PHONE_NUMBER}}, {{lead.is_test}}, {{lead.adgroup_id}}, {{lead.creative_id}}</p>
 </li>
+
+
```
Comment (r0ddy) [inline frontend/src/app/crm/leads/leads.component.html:7]: Why two newlines?

File: frontend/src/app/crm/leads/leads.component.ts [hunk 4]
```diff
@@ -8,6 +8,8 @@ import { Lead } from '../../models/lead.model';
 import { MatTableDataSource } from '@angular/material/table';
 import { MatPaginator } from '@angular/material/paginator';
```
Comment (r0ddy) [inline frontend/src/app/crm/leads/leads.component.ts:10]: Can you delete the new line here?

File: frontend/src/app/crm/leads/leads.component.ts [hunk 5]
```diff
@@ -22,16 +24,25 @@ export class LeadsComponent implements OnInit {
...
-      });
+
+         this.leadService.getAllLeads()
+         .subscribe((leads) => {
+
+          this.dataSource = new MatTableDataSource(leads);
+          this.dataSource.paginator = this.paginator;
+          this.dataSource.sort = this.sort;
+
+   });
```
Comment (r0ddy) [inline frontend/src/app/crm/leads/leads.component.ts:44]: Add indentation every time you enter a new scope/block. So for example here, I would indent once after `getAllLeads(): void {`, and then once more after `.subscribe((leads) => {` . Look here for a better [example](https://google.github.io/styleguide/jsguide.html#formatting-function-expressions).

**Pull request merged**